### ISSUE TITLE
foreach over NonCopyable types

### ISSUE THREAD

**Issue by dan200** [ufcpp/NonCopyableAnalyzer#10]
Trying to write a foreach loop over a NonCopyable collection type causes errors, but logically should not. Here's a (stripped down) example:

```
[NonCopyable]
struct NonCopyableList<T>
{
    public struct Enumerator
    {
        private readonly T[] m_data;
        private int m_index;

        public T Current
        {
            get
            {
                return m_data[m_index];
            }
        }

        public Enumerator(T[] data)
        {
            m_data = data;
            m_index = -1;
        }

        public bool MoveNext()
        {
            return (++m_index < m_data.Length);
        }
    }

    private readonly T[] m_data;

    public NonCopyableList(T[] data)
    {
        m_data = data;
    }

    public Enumerator GetEnumerator()
    {
        return new Enumerator(m_data);
    }
}

public class Program
{
    public static void Main()
    {
        var array = new NonCopyableList<int>(new int[] { 1, 2, 3 });

        var enumerator = array.GetEnumerator(); // No error. Works fine
        while(enumerator.MoveNext())
        {
            var i = enumerator.Current;
            Console.WriteLine(i);
        }

        foreach (var i in array) // ERROR: The type "NonCopyableList" is NonCopyable
        {
            Console.WriteLine(i);
        }
    }
}
```

The latter code block is syntactic sugar for the former, so should not cause compiler errors that the former does not.

Other than this issue, love the analyzer and thanks for writing it :)

**Comment by dan200** [comment]
I have created a pull request with a fix for this: https://github.com/ufcpp/NonCopyableAnalyzer/pull/11

**Comment by ufcpp** [comment]
Published as 0.4.0 https://www.nuget.org/packages/NonCopyableAnalyzer/
